### PR TITLE
Cleanup deprecated metrics

### PR DESCRIPTION
**Pull request by dgrisonnet**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Remove the following deprecated metrics:
- node_collector_evictions_number
- scheduler_e2e_scheduling_duration_seconds

#### Does this PR introduce a user-facing change?

```release-note
Remove the following deprecated metrics:
- node_collector_evictions_number replaced by node_collector_evictions_total
- scheduler_e2e_scheduling_duration_seconds replaced by scheduler_scheduling_attempt_duration_seconds
```

/cc @logicalhan

### CODE REVIEW

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

File: pkg/scheduler/metrics/metrics.go [hunk 1]
```diff
@@ -46,15 +46,6 @@ var (
 			StabilityLevel: metrics.STABLE,
 		}, []string{"result", "profile"})
 
-	e2eSchedulingLatency = metrics.NewHistogramVec(
-		&metrics.HistogramOpts{
-			Subsystem:         SchedulerSubsystem,
-			Name:              "e2e_scheduling_duration_seconds",
-			DeprecatedVersion: "1.23.0",
```
Comment (alculquicondor) [inline pkg/scheduler/metrics/metrics.go:53]: +1

we are 4 versions after.

File: pkg/scheduler/metrics/metrics.go [hunk 2]
```diff
@@ -95,16 +86,6 @@ var (
 			Help:           "Number of pending pods, by the queue type. 'active' means number of pods in activeQ; 'backoff' means number of pods in backoffQ; 'unschedulable' means number of pods in unschedulablePods that the scheduler attempted to schedule and failed; 'gated' is the number of unschedulable pods that the scheduler never attempted to schedule because they are gated.",
 			StabilityLevel: metrics.STABLE,
 		}, []string{"queue"})
-	// SchedulerGoroutines isn't called in some parts where goroutines start.
-	// Goroutines metric replaces SchedulerGoroutines metric. Goroutine metric tracks all goroutines.
-	SchedulerGoroutines = metrics.NewGaugeVec(
-		&metrics.GaugeOpts{
-			Subsystem:         SchedulerSubsystem,
-			DeprecatedVersion: "1.26.0",
```
Comment (alculquicondor) [inline pkg/scheduler/metrics/metrics.go:103]: I would keep this a couple of more versions (so remove in 1.29), just because this metric was quite old.

Comment (dgrisonnet) [reply pkg/scheduler/metrics/metrics.go]: I followed your call on that one and reverted the change.

Just as an fyi, in 1.27 this metric will be hidden by default meaning that it won't be exposed to end-users unless they specifically tell the scheduler to expose it.

**Comment by alculquicondor** [comment]
/approve
for scheduler. Thanks!

File: pkg/scheduler/metrics/metrics.go (hunk 2 above)
Comment (alculquicondor) [reply pkg/scheduler/metrics/metrics.go]: oh is that a new change?

**Review by logicalhan — COMMENTED** [review]
/lgtm

**Pull request merged**